core/iwasm/compilation/aot_emit_gc.c

```diff
@@ -228,6 +259,1213 @@ aot_compile_op_ref_as_non_null(AOTCompContext *comp_ctx,
     return false;
 }
 
+static bool
+aot_call_wasm_struct_obj_new(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                             LLVMValueRef rtt_type, LLVMValueRef *struct_obj)
+{
+    LLVMValueRef param_values[2], func, value, res;
+    LLVMTypeRef param_types[2], ret_type, func_type, func_ptr_type;
+
+    param_types[0] = INT8_PTR_TYPE;
+    param_types[1] = INT8_PTR_TYPE;
+    ret_type = GC_REF_TYPE;
+
+    GET_AOT_FUNCTION(wasm_struct_obj_new, 2);
+
+    /* Call function wasm_struct_obj_new() */
+    param_values[0] = func_ctx->exec_env;
+    param_values[1] = rtt_type;
+    if (!(res = LLVMBuildCall2(comp_ctx->builder, func_type, func, param_values,
+                               2, "call"))) {
+        aot_set_last_error("llvm build call failed.");
+        goto fail;
+    }
+
+    *struct_obj = res;
+    return true;
+fail:
+    return false;
+}
+
+static bool
+aot_struct_obj_set_field(AOTCompContext *comp_ctx, LLVMValueRef struct_obj,
+                         LLVMValueRef field_offset, LLVMValueRef field_value,
+                         uint8 field_size)
+{
+    bool trunc = false;
+    LLVMValueRef field_data_ptr;
+    LLVMTypeRef field_data_type = NULL;
+
+    switch (field_size) {
+        case 1:
+            field_data_type = INT8_TYPE;
+            trunc = true;
+            break;
+        case 2:
+            field_data_type = INT16_TYPE;
+            trunc = true;
+            break;
+        case 4:
+            field_data_type = I32_TYPE;
+            break;
+        case 8:
+            field_data_type = I64_TYPE;
+            break;
+        default:
+            bh_assert(0);
+            break;
+    }
+
+    /* Based on field_size, trunc field_value if necessary */
+    if (trunc) {
+        if (!(field_value =
+                  LLVMBuildTrunc(comp_ctx->builder, field_value,
+                                 field_data_type, "field_value_trunc"))) {
+            aot_set_last_error("llvm build trunc failed.");
+            goto fail;
+        }
+    }
+
+    /* Build field data ptr and store the value */
+    if (!(field_data_ptr = LLVMBuildInBoundsGEP2(
+              comp_ctx->builder, INT8_PTR_TYPE, struct_obj, &field_offset, 1,
+              "field_data_i8p"))) {
+        aot_set_last_error("llvm build gep failed.");
+        goto fail;
+    }
+
+    /* Ensure correct alignment for some platforms when field_size == 8 */
+#if !defined(BUILD_TARGET_X86_64) && !defined(BUILD_TARGET_AMD_64) \
+    && !defined(BUILD_TARGET_X86_32)
```

Comment:
Don't use compiler flags here, wamrc use `--target=xxx` to specify the target.

Seems no need to add check and cast to INT32_PTR_TYPE here, remove this code piece. The field data is already n-bytes aligned in non-x86 targets.